### ISSUE TITLE
Keyboard support for actions on template variables

### ISSUE THREAD

**Issue by hauber** [RealRaven2000/SmartTemplate4#63]
It would be nice to interact with template variables not only by mouse but using the keyboard:
![image](https://user-images.githubusercontent.com/1751651/82121153-30036a80-978b-11ea-8662-2cc46779ff90.png)

**Comment by RealRaven2000** [comment]
Good idea! any ideas on what keys to use and how to focus on a field? TAB is currently captured for inserting TAB characters, but maybe something like CTRL+TAB or similar could be used to open the first field. This could just open up the menu (as shown above) and maybe we could add accelerator keys. 

I think we are also missing a fourth option "type manually". This one could just replace the whole field with the plain typed text, permanently?

**Comment by hauber** [comment]
Ctrl+Tab is already taken. I haven't looked at all the current mappings, but probably two modifier keys are safer, like Ctrl+Shift+T. It could jump to the next template variable and open its context menu. The context menu can already be navigated using the arrow keys.

Would the fourth option that you mention be equivalent to removing the variable (= thrid option) and entering the text? Then we wouldn't need another option. Just make sure that after removing the variable, the cursor is placed in the variable's position.

**Comment by RealRaven2000** [comment]
> Would the fourth option that you mention be equivalent to removing the variable (= thrid option) and entering the text? Then we wouldn't need another option. Just make sure that after removing the variable, the cursor is placed in the variable's position.

yes I would just position an editable field within the variable (and possibly change its shape / color) where you can start typing:

![image](https://user-images.githubusercontent.com/6314674/82131521-d2dcd880-97cd-11ea-88e3-f57de5d76e3c.png)

e.g. by highlighting the placeholder text. Once you start it will immediately remove the complete field and you will be placed there with the writing cursor.

**Comment by hauber** [comment]
"Just typing" to replace a field sounds like the right approach.

My use case is rather quick and lightweight, so the lack of undo/save is not critical.

**Comment by RealRaven2000** [comment]
> "Just typing" to replace a field sounds like the right approach.
> 
> My use case is rather quick and lightweight, so the lack of undo/save is not critical.

definitely agree - lightweight is the best way to go. SmartTemplate4 is already terribly complicated to set up. Once it's done it's pretty easy to use the templates in every day usage. It's probably my number 1 time saver for writing support mails for it. 

![image](https://user-images.githubusercontent.com/6314674/82141391-4f52d400-982d-11ea-8d73-3b9a5bc30f17.png)
